lib/db_utils.go

```diff
@@ -10661,6 +10681,409 @@ func DBDeletePostAssociationWithTxn(txn *badger.Txn, snap *Snapshot, association
 	return nil
 }
 
+// -------------------------------------------------------------------------------------
+// Lockup DB Operations
+// -------------------------------------------------------------------------------------
+
+// LockedBalanceEntry DB Key Operations
+
+func _dbKeyForLockedBalanceEntry(lockedBalanceEntry LockedBalanceEntry) []byte {
+	key := append([]byte{}, Prefixes.PrefixLockedBalanceEntryByHODLerPKIDProfilePKIDExpirationTimestampNanoSecs...)
+	key = append(key, lockedBalanceEntry.HODLerPKID[:]...)
+	key = append(key, lockedBalanceEntry.ProfilePKID[:]...)
+	return append(key, EncodeUint64(uint64(lockedBalanceEntry.ExpirationTimestampNanoSecs))...)
+}
+
+func DBPrefixKeyForLockedBalanceEntryByHODLerPKIDandProfilePKID(lockedBalanceEntry *LockedBalanceEntry) []byte {
+	data := append([]byte{}, Prefixes.PrefixLockedBalanceEntryByHODLerPKIDProfilePKIDExpirationTimestampNanoSecs...)
+	data = append(data, lockedBalanceEntry.HODLerPKID.ToBytes()...)
+	data = append(data, lockedBalanceEntry.ProfilePKID.ToBytes()...)
+	return data
+}
+
+// LockedBalanceEntry Put/Delete Operations (Badger Writes)
+
+func DbPutLockedBalanceEntryMappingsWithTxn(txn *badger.Txn, snap *Snapshot, blockHeight uint64,
+	lockedBalanceEntry LockedBalanceEntry) error {
+	// Sanity check the fields in the LockedBalanceEntry used in constructing the key.
+	if len(lockedBalanceEntry.HODLerPKID) != btcec.PubKeyBytesLenCompressed {
+		return fmt.Errorf("DbPutLockedBalanceEntryMappingsWithTxn: HODLer PKID "+
+			"length %d != %d", len(lockedBalanceEntry.HODLerPKID), btcec.PubKeyBytesLenCompressed)
+	}
+	if len(lockedBalanceEntry.ProfilePKID) != btcec.PubKeyBytesLenCompressed {
+		return fmt.Errorf("DbPutLockedBalanceEntryMappingsWithTxn: Profile PKID "+
+			"length %d != %d", len(lockedBalanceEntry.ProfilePKID), btcec.PubKeyBytesLenCompressed)
+	}
+
+	if err := DBSetWithTxn(txn, snap, _dbKeyForLockedBalanceEntry(lockedBalanceEntry),
```

Comment:
Good job using DBSetWithTxn. I was worried you would use txn.Set() which causes bugs because it doesn't update the snapshot properly for hypersync.